flusher.go

```diff
@@ -17,20 +18,47 @@ import (
 // Flush takes the slices of metrics, combines then and marshals them to json
 // for posting to Datadog.
 func (s *Server) Flush(interval time.Duration, metricLimit int) {
-	// number of ddmetrics generated when a histogram flushes
-	histogramSize := 3 + len(s.HistogramPercentiles)
+	percentiles := s.HistogramPercentiles
+	if s.ForwardAddr != "" {
+		// don't publish percentiles if we're going to forward
+		percentiles = nil
+	}
 
 	// allocating this long array to count up the sizes is cheaper than appending
 	// the []DDMetrics together one at a time
 	tempMetrics := make([]WorkerMetrics, 0, len(s.Workers))
-	totalLength := 0
+	var (
+		totalCounters   int
+		totalGauges     int
+		totalHistograms int
+		totalSets       int
+		totalTimers     int
+
+		totalLocalHistograms int
+		totalLocalSets       int
+		totalLocalTimers     int
+	)
 	for i, w := range s.Workers {
 		s.logger.WithField("worker", i).Debug("Flushing")
 		wm := w.Flush()
 		tempMetrics = append(tempMetrics, wm)
-		totalLength += len(wm.counters) + len(wm.gauges) + len(wm.sets) + ((len(wm.timers) + len(wm.histograms)) * histogramSize)
+
+		totalCounters += len(wm.counters)
+		totalGauges += len(wm.gauges)
+		totalHistograms += len(wm.histograms)
+		totalSets += len(wm.sets)
+		totalTimers += len(wm.timers)
+
+		totalLocalHistograms += len(wm.localHistograms)
+		totalLocalSets += len(wm.localSets)
+		totalLocalTimers += len(wm.localTimers)
 	}
 
+	totalLength := totalCounters + totalGauges + (totalTimers+totalHistograms)*(3+len(percentiles)) +
+		totalLocalSets + (totalLocalTimers+totalLocalHistograms)*(3+len(s.HistogramPercentiles))
+	if s.ForwardAddr == "" {
+		totalLength += totalSets
```

Comment:
Why this is just `totalSets`? Isn't `totalHistograms` gonna be a thing here too?

Think it would be good to have a comment here describing why this is happening?